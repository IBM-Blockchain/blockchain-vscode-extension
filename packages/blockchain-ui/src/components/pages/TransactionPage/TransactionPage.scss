@import '@carbon/colors/scss/colors';
@import '../../../mixins/layout-mixins.scss';

.page-container {
    @include default-page-styling;

    .titles-container {
        padding: 0px;

        h2 {
            padding: $spacing-06 0px;
        }

        .breadcrumb {
            font-size: 0.875rem;
        }

        .breadcrumb .breadcrumb-divider {
            margin: 0 $spacing-03;
        }

        .breadcrumb, .breadcrumb .breadcrumb-divider {
            color: $gray-10;
            margin: 0 $spacing-03;
        }
    }

    .bx--form {
        .bx--dropdown__wrapper, .bx--form-item, .bx--multi-select__wrapper, .associate-tx-data {
            margin: $spacing-06 0;
        }
    }
}


.vscode-light {
    .page-container {
        .titles-container {
            .breadcrumb, .breadcrumb .breadcrumb-divider {
                color: $gray-70;
            }
        }
    }
}
