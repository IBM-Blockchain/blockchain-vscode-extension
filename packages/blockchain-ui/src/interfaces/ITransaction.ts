interface ITransaction {
    name: string;
    parameters: Array<{ description: string, name: string, schema: {} }>;
    returns: { type: string };
    tag: Array<string>;
}

export default ITransaction;
