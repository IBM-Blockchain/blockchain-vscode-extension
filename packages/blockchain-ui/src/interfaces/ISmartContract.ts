import ITransaction from './ITransaction';

interface ISmartContract {
    name: string;
    version: string;
    channel: string;
    label: string;
    transactions: Array<ITransaction>;
    namespace: string;
    peerNames: string[];
}

export default ISmartContract;
