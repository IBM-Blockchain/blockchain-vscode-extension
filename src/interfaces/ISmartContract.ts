import ITransaction from './ITransaction';

interface ISmartContract {
    name: string;
    version: string;
    channel: string;
    label: string;
    transactions: Array<ITransaction>;
    namespace: string;
}

export default ISmartContract;
