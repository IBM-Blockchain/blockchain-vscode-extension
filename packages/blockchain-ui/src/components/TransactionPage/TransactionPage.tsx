import React, { Component } from 'react';
import './TransactionPage.scss';
import TransactionForm from '../TransactionForm/TransactionForm';
import TransactionOutput from '../TransactionOutput/TransactionOutput';
import ISmartContract from '../../interfaces/ISmartContract';

interface PageProps {
    gatewayName: string;
    smartContract: ISmartContract;
    transactionOutput: string;
    postMessageHandler: (command: string, data?: any) => void;
}

interface PageState {
    gatewayName: string;
    smartContract: ISmartContract;
    transactionOutput: string;
    postMessageHandler: (command: string, data?: any) => void;
}

class TransactionPage extends Component<PageProps, PageState> {
    constructor(props: Readonly<PageProps>) {
        super(props);
        this.state = {
            gatewayName: this.props.gatewayName,
            smartContract: this.props.smartContract,
            transactionOutput: this.props.transactionOutput,
            postMessageHandler: this.props.postMessageHandler
        };
    }

    componentDidUpdate(prevProps: PageProps): void {
        if (prevProps.transactionOutput !== this.props.transactionOutput) {
            this.setState({
                transactionOutput: this.props.transactionOutput
            });
        }
    }

    render(): JSX.Element {
        return (
            <div className='page-container bx--grid' data-test-id='txn-page'>
                <div className='inner-container bx--row'>
                    <div className='page-contents bx--col'>
                        <div className='titles-container'>
                            <span className='home-link'>Transacting with: {this.state.gatewayName} > {this.state.smartContract.channel} > {this.state.smartContract.label}</span>
                            <h2>Create a new transaction</h2>
                        </div>
                        <div className='contents-container bx--row'>
                            <div className='bx--col'>
                                <TransactionForm smartContract={this.state.smartContract} postMessageHandler={this.state.postMessageHandler}/>
                            </div>
                        </div>
                    </div>
                    <div className='bx--col'>
                        <TransactionOutput output={this.state.transactionOutput}/>
                    </div>
                </div>
            </div>
        );
    }

}

export default TransactionPage;
