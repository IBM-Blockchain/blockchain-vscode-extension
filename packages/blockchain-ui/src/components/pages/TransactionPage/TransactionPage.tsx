import React, { Component } from 'react';
import './TransactionPage.scss';
import TransactionForm from '../../elements/TransactionForm/TransactionForm';
import TransactionOutput from '../../elements/TransactionOutput/TransactionOutput';
import ISmartContract from '../../../interfaces/ISmartContract';

interface IProps {
    transactionData: {gatewayName: string, smartContract: ISmartContract };
    transactionOutput: string;
}

interface IState {
    gatewayName: string;
    smartContract: ISmartContract;
    transactionOutput: string;
}

class TransactionPage extends Component<IProps, IState> {
    constructor(props: Readonly<IProps>) {
        super(props);
        this.state = {
            gatewayName: this.props.transactionData.gatewayName,
            smartContract: this.props.transactionData.smartContract,
            transactionOutput: this.props.transactionOutput,
        };
    }

    componentDidUpdate(prevProps: IProps): void {
        if (prevProps.transactionOutput !== this.props.transactionOutput) {
            this.setState({
                transactionOutput: this.props.transactionOutput
            });
        }
    }

    render(): JSX.Element {
        return (
            <div className='page-container bx--grid' data-test-id='txn-page'>
                <div className='inner-container bx--row'>
                    <div className='page-contents bx--col'>
                        <div className='titles-container'>
                            <span className='home-link'>Transacting with: {this.state.gatewayName} > {this.state.smartContract.channel} > {this.state.smartContract.label}</span>
                            <h2>Create a new transaction</h2>
                        </div>
                        <div className='contents-container bx--row'>
                            <div className='bx--col'>
                                <TransactionForm smartContract={this.state.smartContract}/>
                            </div>
                        </div>
                    </div>
                    <div className='bx--col'>
                        <TransactionOutput output={this.state.transactionOutput}/>
                    </div>
                </div>
            </div>
        );
    }

}

export default TransactionPage;
