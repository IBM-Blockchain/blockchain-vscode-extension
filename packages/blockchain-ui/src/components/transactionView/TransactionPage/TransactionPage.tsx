import React, { Component } from 'react';
import './TransactionPage.scss';
import TransactionForm from '../TransactionForm/TransactionForm';
import TransactionOutput from '../TransactionOutput/TransactionOutput';
import ISmartContract from '../../../interfaces/ISmartContract';

interface IProps {
    gatewayName: string;
    smartContract: ISmartContract;
    transactionOutput: string;
    postMessageHandler: (command: string, data?: any) => void;
}

interface IState {
    gatewayName: string;
    smartContract: ISmartContract;
    transactionOutput: string;
    postMessageHandler: (command: string, data?: any) => void;
}

class TransactionPage extends Component<IProps, IState> {
    constructor(props: Readonly<IProps>) {
        super(props);
        this.state = {
            gatewayName: this.props.gatewayName,
            smartContract: this.props.smartContract,
            transactionOutput: this.props.transactionOutput,
            postMessageHandler: this.props.postMessageHandler
        };
    }

    componentDidUpdate(prevProps: IProps): void {
        if (prevProps.transactionOutput !== this.props.transactionOutput) {
            this.setState({
                transactionOutput: this.props.transactionOutput
            });
        }
    }

    render(): JSX.Element {
        return (
            <div className='page-container bx--grid' data-test-id='txn-page'>
                <div className='inner-container bx--row'>
                    <div className='page-contents bx--col'>
                        <div className='titles-container'>
                            <span className='home-link'>Transacting with: {this.state.gatewayName} > {this.state.smartContract.channel} > {this.state.smartContract.label}</span>
                            <h2>Create a new transaction</h2>
                        </div>
                        <div className='contents-container bx--row'>
                            <div className='bx--col'>
                                <TransactionForm smartContract={this.state.smartContract} postMessageHandler={this.state.postMessageHandler}/>
                            </div>
                        </div>
                    </div>
                    <div className='bx--col'>
                        <TransactionOutput output={this.state.transactionOutput}/>
                    </div>
                </div>
            </div>
        );
    }

}

export default TransactionPage;
