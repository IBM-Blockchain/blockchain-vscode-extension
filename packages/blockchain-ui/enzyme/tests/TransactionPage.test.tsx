// tslint:disable no-unused-expression
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import TransactionPage from '../../src/components/transactionView/TransactionPage/TransactionPage';
import chai from 'chai';
import sinon from 'sinon';
import sinonChai from 'sinon-chai';
import ITransaction from '../../src/interfaces/ITransaction';
import ISmartContract from '../../src/interfaces/ISmartContract';
chai.should();
chai.use(sinonChai);

describe('TransactionPage component', () => {
    let mySandBox: sinon.SinonSandbox;
    let postMessageHandlerStub: sinon.SinonStub;

    const transactionOne: ITransaction = {
        name: 'transactionOne',
        parameters: [{
            description: '',
            name: 'name',
            schema: {}
        }],
        returns: {
            type: ''
        },
        tag: ['submit']
    };

    const transactionTwo: ITransaction = {
        name: 'transactionTwo',
        parameters: [],
        returns: {
            type: ''
        },
        tag: ['submit']
    };

    const greenContract: ISmartContract = {
        name: 'greenContract',
        version: '0.0.1',
        channel: 'mychannel',
        label: 'greenContract@0.0.1',
        transactions: [transactionOne, transactionTwo],
        namespace: 'GreenContract'
    };

    const mockTransactionOutput: string = 'here is some transaction output';
    const moreMockTransactionOutput: string = 'here is some more transaction output';

    beforeEach(async () => {
        mySandBox = sinon.createSandbox();
        postMessageHandlerStub = mySandBox.stub();
    });

    afterEach(async () => {
        mySandBox.restore();
    });

    it('should render the expected snapshot', async () => {
        const component: any = renderer
            .create(<TransactionPage gatewayName={'my gateway'} smartContract={greenContract} transactionOutput={mockTransactionOutput} postMessageHandler={postMessageHandlerStub}/>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should update the transaction output when something new is passed down through props', async () => {
        const componentDidUpdateSpy: sinon.SinonSpy = mySandBox.spy(TransactionPage.prototype, 'componentDidUpdate');
        const component: any = mount(<TransactionPage gatewayName={'my gateway'} smartContract={greenContract} transactionOutput={mockTransactionOutput} postMessageHandler={postMessageHandlerStub}/>);
        component.state().transactionOutput.should.equal(mockTransactionOutput);

        component.setProps({
            transactionOutput: moreMockTransactionOutput
        });
        componentDidUpdateSpy.should.have.been.called;
        component.state().transactionOutput.should.equal(moreMockTransactionOutput);
    });
});
