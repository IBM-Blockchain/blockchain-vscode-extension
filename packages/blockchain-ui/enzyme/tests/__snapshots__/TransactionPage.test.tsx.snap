// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionPage component should render the expected snapshot 1`] = `
<div
  className="page-container bx--grid"
  data-test-id="txn-page"
>
  <div
    className="inner-container bx--row"
  >
    <div
      className="page-contents bx--col"
    >
      <div
        className="titles-container"
      >
        <span
          className="home-link"
        >
          Transacting with: 
          my gateway
           &gt; 
          mychannel
           &gt; 
          greenContract@0.0.1
        </span>
        <h2>
          Create a new transaction
        </h2>
      </div>
      <div
        className="contents-container bx--row"
      >
        <div
          className="bx--col"
        >
          <form
            className="bx--form"
            id="create-txn-form"
          >
             
            <fieldset
              className="bx--fieldset"
            >
              <legend
                className="bx--label"
              >
                Transaction label
              </legend>
              <div
                className="bx--row inline-row hide-label"
              >
                <div
                  className="bx--form-item bx--text-input-wrapper"
                >
                  <label
                    className="bx--label bx--visually-hidden"
                    htmlFor="transaction-label-input"
                  >
                    Enter transaction label
                  </label>
                  <div
                    className="bx--text-input__field-wrapper"
                    data-invalid={null}
                  >
                    <input
                      className="bx--text-input bx--text__input"
                      disabled={false}
                      id="transaction-label-input"
                      onChange={[Function]}
                      onClick={[Function]}
                      placeholder="Enter transaction label"
                      title="Enter transaction label"
                      type="text"
                    />
                  </div>
                </div>
                <button
                  className="bx--btn bx--btn--field bx--btn--primary"
                  disabled={false}
                  id="save-txn-button"
                  tabIndex={0}
                  type="button"
                >
                  Save
                </button>
              </div>
            </fieldset>
            <fieldset
              className="bx--fieldset"
            >
              <legend
                className="bx--label"
              >
                Transaction name
              </legend>
              <div
                className="bx--form-item"
              >
                <div
                  className="bx--select select-width"
                >
                  <label
                    className="bx--label"
                    htmlFor="transaction-name-select"
                  >
                    Transaction name*
                  </label>
                  <div
                    className="bx--select-input__wrapper"
                    data-invalid={null}
                  >
                    <select
                      className="bx--select-input"
                      id="transaction-name-select"
                      onChange={[Function]}
                    >
                      <option
                        className="bx--select-option"
                        disabled={false}
                        hidden={true}
                        value="placeholder-item"
                      >
                        Select the transaction name
                      </option>
                      <option
                        className="bx--select-option"
                        disabled={false}
                        hidden={false}
                        value="transactionOne"
                      >
                        transactionOne
                      </option>
                      <option
                        className="bx--select-option"
                        disabled={false}
                        hidden={false}
                        value="transactionTwo"
                      >
                        transactionTwo
                      </option>
                    </select>
                    <svg
                      aria-hidden={true}
                      className="bx--select__arrow"
                      focusable="false"
                      height={16}
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 16 16"
                      width={16}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <polygon
                        points="8,11 3,6 3.7,5.3 8,9.6 12.3,5.3 13,6"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </fieldset>
            <fieldset
              className="bx--fieldset"
              id="target-peer-input"
            >
              <legend
                className="bx--label"
              >
                Peer targeting
              </legend>
              <div
                className="bx--form-item bx--checkbox-wrapper"
              >
                <input
                  className="bx--checkbox"
                  id="target-peer-checkbox"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="bx--checkbox-label"
                  htmlFor="target-peer-checkbox"
                  title={null}
                >
                  <span
                    className="bx--checkbox-label-text"
                  >
                    Target custom peer
                  </span>
                </label>
              </div>
              <div
                className="bx--form-item"
              >
                <div
                  className="bx--select select-width hide-label"
                >
                  <label
                    className="bx--label"
                    htmlFor="peers-select"
                  >
                    Peers
                  </label>
                  <div
                    className="bx--select-input__wrapper"
                    data-invalid={null}
                  >
                    <select
                      className="bx--select-input"
                      id="peers-select"
                    >
                      <option
                        className="bx--select-option"
                        disabled={false}
                        hidden={false}
                        value="select-a-peer"
                      >
                        Select a peer
                      </option>
                    </select>
                    <svg
                      aria-hidden={true}
                      className="bx--select__arrow"
                      focusable="false"
                      height={16}
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 16 16"
                      width={16}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <polygon
                        points="8,11 3,6 3.7,5.3 8,9.6 12.3,5.3 13,6"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </fieldset>
            <fieldset
              className="bx--fieldset"
            >
              <legend
                className="bx--label"
              >
                Arguments
              </legend>
              <div
                className="bx--form-item"
              >
                <label
                  className="bx--label"
                  htmlFor="arguments-text-area"
                >
                  Arguments*
                </label>
                <div
                  className="bx--text-area__wrapper"
                  data-invalid={null}
                >
                  <textarea
                    aria-describedby={null}
                    aria-invalid={null}
                    className="bx--text-area"
                    cols={50}
                    disabled={false}
                    id="arguments-text-area"
                    onChange={[Function]}
                    onClick={[Function]}
                    placeholder=""
                    rows={4}
                    value=""
                  />
                </div>
              </div>
            </fieldset>
            <fieldset
              className="bx--fieldset"
            >
              <legend
                className="bx--label"
              >
                Transient data
              </legend>
              <div
                className="bx--form-item bx--text-input-wrapper"
              >
                <label
                  className="bx--label"
                  htmlFor="transient-data-input"
                >
                  Transient data (optional) - e.g. {"key": "value"}
                </label>
                <div
                  className="bx--text-input__field-wrapper"
                  data-invalid={null}
                >
                  <input
                    className="bx--text-input bx--text__input"
                    disabled={false}
                    id="transient-data-input"
                    onChange={[Function]}
                    onClick={[Function]}
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </fieldset>
            <fieldset
              className="bx--fieldset"
              id="submit-and-evaluate-buttons"
            >
              <legend
                className="bx--label"
              >
                Submit and Evaluate buttons
              </legend>
              <button
                className="submit-and-evaluate-buttons bx--btn bx--btn--field bx--btn--primary bx--btn--disabled"
                disabled={true}
                id="evaluate-button"
                onClick={[Function]}
                tabIndex={0}
                type="button"
              >
                Evaluate
              </button>
              <button
                className="submit-and-evaluate-buttons bx--btn bx--btn--field bx--btn--primary bx--btn--disabled"
                disabled={true}
                id="submit-button"
                onClick={[Function]}
                tabIndex={0}
                type="button"
              >
                Submit
              </button>
            </fieldset>
             
          </form>
        </div>
      </div>
    </div>
    <div
      className="bx--col"
    >
      <div
        className="output-panel"
        id="output-panel"
      >
        <p
          className="output-title"
        >
          Output for untitled transaction
        </p>
        <p
          className="output-body"
        >
          here is some transaction output
        </p>
      </div>
    </div>
  </div>
</div>
`;
